src/wp-admin/includes/image.php

```diff
@@ -393,79 +439,114 @@ function _wp_make_subsizes( $new_sizes, $file, $image_meta, $attachment_id ) {
 		return array();
 	}
 
-	// Check if any of the new sizes already exist.
-	if ( isset( $image_meta['sizes'] ) && is_array( $image_meta['sizes'] ) ) {
-		foreach ( $image_meta['sizes'] as $size_name => $size_meta ) {
-			/*
-			 * Only checks "size name" so we don't override existing images even if the dimensions
-			 * don't match the currently defined size with the same name.
-			 * To change the behavior, unset changed/mismatched sizes in the `sizes` array in image meta.
-			 */
-			if ( array_key_exists( $size_name, $new_sizes ) ) {
-				unset( $new_sizes[ $size_name ] );
-			}
-		}
-	} else {
-		$image_meta['sizes'] = array();
+	// Set up a top level sources array for the full size images.
+	if ( ! isset( $metadata['sources'] ) || ! is_array( $metadata['sources'] ) ) {
+		$metadata['sources'] = array();
 	}
+	$mime_type = get_post_mime_type( $attachment_id );
```

Comment:
In `wp_create_image_subsizes()` above we use `$imagesize['mime']` which is based on the given file, so it's a bit odd to rely on `get_post_mime_type` here. This is especially problematic related to my above comment, we need to ensure this supports the `application/pdf` special case. I'd suggest something like the following:
```suggestion
	$mime_type = wp_get_image_mime( $file );
```

Since the `$file` provided to this function in the `application/pdf` case is already the `full` JPEG file, that'll fix this issue and also support generating WebP versions for this case.

Reply:
switched to this

Reply:
The latest PR is still using `get_post_mime_type()` in one critical place.